code/object/objcollide.cpp

```diff
@@ -1043,8 +1043,8 @@ void obj_sort_and_collide()
 	if ( !(Game_detail_flags & DETAIL_FLAG_COLLISION) )
 		return;
 
-	SCP_vector<int> sort_list_y;
-	SCP_vector<int> sort_list_z;
+	static SCP_vector<int> sort_list_y;
+	static SCP_vector<int> sort_list_z;
```

Comment:
These should be actual global variables to make sure that the "cost" of the static variables is clearly visible.

Reply:
The "cost" doesn't need to be visible.  The cost is plainly apparent from the `static` keyword (and anyway, it is hardly a cost since it speeds up the function).

A much better reason for not making them global variables is that they are only used in the one function, so they should remain entirely within the scope of the one function.

Reply:
The "cost" is the maintainability cost since it makes it harder to see where in a module there is global data.

Also, there definitely is a constant cost of static variables for every function call since the code needs to check if the variable has already been initialized and that needs to happen with an atomic instruction so the performance isn't the best. If this is moved to a "normal" static, global variable then there are no hidden costs.

Reply:
FWIW I agree with @asarium here; the performance difference is small, but not nonexistent, and since it's per call the savings add up. If we can gain a few nanoseconds per function call "for free", why not?